libs/geoservices/covid/src/lib/pages/maps/components/lockdown-map/lockdown-map.component.ts

```diff
@@ -0,0 +1,258 @@
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { forkJoin, BehaviorSubject } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
+
+@Component({
+  selector: 'tamu-gisc-lockdown-map',
+  templateUrl: './lockdown-map.component.html',
+  styleUrls: ['./lockdown-map.component.scss']
+})
+export class LockdownMapComponent implements OnInit {
+  constructor(private mapService: MapboxMapService, private http: HttpClient) {}
+  public unformattedData;
+  public statData: Array<StatRecord>;
+  public lockdownButtonToggle = false;
+  public infoBoxModel: BehaviorSubject<IInfoBox> = new BehaviorSubject(undefined);
+  public countyBoxModel: BehaviorSubject<CountyStateBox> = new BehaviorSubject(undefined);
+  public ngOnInit(): void {
+    this.mapService.loaded.subscribe((map) => {
+      this.getStats();
+      const zoomThreshold = 3;
+
+      map.addSource('county-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.countylines'
+      });
+
+      map.addSource('state-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.statelines'
+      });
+
+      map.addLayer(
+        {
+          id: 'counties-lockdowns',
+          type: 'fill',
+          source: 'county-lines',
+          layout: {
+            visibility: 'none'
+          },
+          'source-layer': 'county-lines',
+          paint: {
+            'fill-outline-color': '#303030'
+          }
+        },
+        'road-motorway-trunk'
+      );
+
+      map.addLayer(
+        {
+          id: 'counties-claims',
+          type: 'fill',
+          source: 'county-lines',
+          'source-layer': 'county-lines',
+          paint: {
+            'fill-outline-color': '#FFFFFF'
+          }
+        },
+        'counties-lockdowns'
+      );
+
+      map.addLayer(
+        {
+          id: 'statelines',
+          type: 'line',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          paint: {
+            'line-color': '#000000',
+            'line-width': 1
+          }
+        },
+        'state-label'
+      );
+
+      map.on('mousemove', 'counties-lockdowns', (e) => {
+        const feature = e.features[0];
+        this.countyBoxModel.next({
+          county: feature.properties.NAME
+        });
+      });
+
+      map.on('click', 'counties-lockdowns', (e) => {
+        const feature = e.features[0];
+        const hoveredCounty = this.statData.filter((county) => county.fips === feature.properties.fips);
+        const updatedDate: Date = hoveredCounty[0]['lockdownInfo'][0]
+          ? hoveredCounty[0]['lockdownInfo'][0]['updated']
+          : null;
+        const createdDate: Date = hoveredCounty[0]['lockdownInfo'][0]
+          ? hoveredCounty[0]['lockdownInfo'][0]['created']
+          : null;
```

Comment:
This is one of the reasons why it event types should be used. The `click` event *should* be a generic, which would enable typescript to know what kind of features are coming back in the click event object which in turn allows you to use direct property accessors with dot notation (strongly typed) instead of instead of bracket notation (weakly typed, explicit `any`).

In the event that lockdown info interface changed, you'd see these as errors at build-time instead of run-time.

Reply:
Marked resolved accidentally

Reply:
removed bracket notation and used dot notation wherever applicable on the object.